File: CHECK-IF-DOUBLE/cid.go
package main

import (
	"fmt"
)

func checkIfExist(arr []int) bool {
	set := make(map[int]bool)

	for _, val := range arr {
		if set[val*2] {
			return true
		}
		if val%2 == 0 && set[val/2] {
			return true
		}
		set[val] = true
	}
	return false
}

func main() {
	arr := []int{10, 2, 5, 3}
	fmt.Println(checkIfExist(arr)) 
}

File: CHECK-IF-DOUBLE/cid.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        s = set(arr)
        # for i in range(len(arr)):
        #     num = arr[i]/2
        #     for j in range (len(arr)):
        #         if arr[j] == num and j != i: 
        #             return True
        # for i in range(len(arr)):
        #     num = arr[i]/2
        #     if num in arr and i != arr.index(num):
        #         return True
            
        # return False
        s = set()
        for i in arr:
            if i/2 in s or i*2 in s:
                return True
            s.add(i)
            
        return False
        